llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -21962,6 +21962,35 @@ SDValue tryLowerPartialReductionToWideAdd(SDNode *N,
   return DAG.getNode(TopOpcode, DL, AccVT, BottomNode, ExtOp);
 }
 
+static SDValue foldRevInvolution(SDNode *N) {
+  SDValue InnerRev = N->getOperand(1);
+  if (!InnerRev.hasOneUse())
+    return SDValue();
```

Comment:
Does it matter if the inner rev has more than one use? This only replaces the outer rev, so if there's another use of the inner rev it should be preserved.